### ISSUE TITLE
error decoding map: key is not a valid map index

### ISSUE THREAD

**Issue by bj0** [orchestr7/ktoml#287]
I tried this:

```kotlin
val d = mapOf("key" to "value")
val t = Toml.encodeToString(d)

Toml.decodeFromString<Map<String,String>>(t)
```
It throws an `IllegalArgumentException: key is not a valid map index`.

This seems so simple that maybe I'm using it wrong?

It works fine with `Json` serialization.

**Comment by matytyma** [comment]
What does `t` look like?

**Comment by bj0** [comment]
it was `key = "value"`

**Comment by matytyma** [comment]
Hmmm doesn't `encodeToString` produce any extra newlines etc.? Did you try `Toml.decodeFromString<Map<String,String>>("""key = "value"""")`?

**Comment by bj0** [comment]
I'm doing this in a kotlin scratch file, if that matters.

```
val d = mapOf("key" to "value")
val t = Toml.encodeToString(d)
println("encoded: '$t'")

try {
    Toml.decodeFromString<Map<String, String>>("""key = "value"""")
} catch (t: Throwable) {
    t.toString()
}
```
output is `encoded: 'key = "value"'` and the same exception is thrown on decode.

**Comment by matytyma** [comment]
That definitely is a bug in the decoding process, I'd have to look at the source in an IDE to see where it is thrown from. I haven't used ktoml for a while as I've hit a roadblock bug and had to migrate to tomlkt...

**Comment by orchestr7** [comment]
Yeah, right now I am very lazily support this project, hope I will have some time to return back to it

**Comment by matytyma** [comment]
Sure, I'll dig into it once I find some free time

**Comment by orchestr7** [comment]
Should be fixed in #291. Will see in release
